### PR TITLE
Abdul/create post api

### PR DESCRIPTION
**Pull request by abdul-tal**
added create_post api

### CODE REVIEW

File: app/consts/uri.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const localstackHost = 'http://localhost:4566';
```
Comment (kpahwani) [inline app/consts/uri.js]: Could this be added to the default config?

File: app/controllers/postController.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const Posts = require('../models/posts');
+const { uploadFile } = require('../helpers/aws/uploader');
+const { S3_BUCKET_NAME: bucketName } = require('config');
+
+
+const createPost = async (data, file) => {
+    console.log('CreatePost Data Controller...');
```
Comment (kpahwani) [inline app/controllers/postController.js]: restrict to use console logger, instead, use either centralized or file-based logging.

File: app/helpers/aws/s3Client.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    return {
+        region,
+        accessKeyId,
+        secretAccessKey
+    };
+};
+const conf = getAWSCredentialsOptions();
+const s3client = new S3Client(conf);
+
+module.exports = s3client;
```
Comment (kpahwani) [inline app/helpers/aws/s3Client.js]: Add a new line.

File: app/models/posts.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  createdAt: {
+    type: String,
+    required: true,
+    immutable: true,
+    default: () => Date.now()
+  },
+  updatedAt: {
+    type: String,
+    required: true,
+    default: () => Date.now()
```
Comment (kpahwani) [inline app/models/posts.js:34]: Add **deleted_at** attribute.

File: app/routes/posts.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+const postRoutes = require('express').Router();
+const multer  = require('multer')
+const upload = multer({ dest: 'uploads/' })
+const postController = require('../controllers/postController')
+
+
+postRoutes.post('/create_post', upload.single('mediaFile'), async (req, res) => {
```
Comment (kpahwani) [inline app/routes/posts.js]: should be **post** is of create_post. The method explains the operation.

File: config/default.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+    "AWS_REGION": "us-west-2",
+    "S3_BUCKET_NAME": "posts-bucket",
+    "databaseConString": "mongodb:/127.0.0.1:27017/insta_db_local"
```
Comment (kpahwani) [inline config/default.json]: Follow upper case for **configs**.

File: config/default.json [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+}
```
Comment (kpahwani) [inline config/default.json]: Add new line.

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    ports:
+      - '4563-4599:4563-4599'
+      - '8055:8080'
+    environment:
+      - SERVICES=s3
+      - DEBUG=1
+      - DATA_DIR=/tmp/localstack/data
+    volumes:
+      - './.localstack:/tmp/localstack'
+      - '/var/run/docker.sock:/var/run/docker.sock'
```
Comment (kpahwani) [inline docker-compose.yml]: Add a new line.

File: index.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+const { server } = require('./app/server');
+
+server().then((app) => {
+    const port = process.env.ISSUE_TO_BCF_PORT || 8002;
```
Comment (kpahwani) [inline index.js]: Invalid variable **process.env.ISSUE_TO_BCF_PORT**

**Review by kpahwani — CHANGES_REQUESTED** [review]
Left a few comments.

File: app/controllers/postController.js (hunk 2 above)
Comment (abdul-tal) [reply app/controllers/postController.js]: It will be updated in further commits

File: app/routes/posts.js (hunk 5 above)
Comment (abdul-tal) [reply app/routes/posts.js]: you mean the api should be /post, right?

**Review by kpahwani — APPROVED** [review]

**Pull request merged**